include/wex/util.hpp

```diff
@@ -10,8 +11,65 @@ class Pinned {
 	Pinned(Pinned const&) = delete;
 	Pinned(Pinned&)		  = delete;
 
+	Pinned operator=(Pinned&&) = delete;
+	Pinned operator=(Pinned const&) = delete;
+
   public:
 	Pinned() = default;
 };
 
+#define WEX_ENSURE(cond) ((cond) ? (void(0)) : std::terminate())
```

Comment:
Does terminating display the appropriate error message? Might want to use `assert` otherwise

Reply:
`assert` is debug only right? I was wondering `NotNullPtr` should make this assertion only in debug mode or in release mode as well.

Reply:
It's fair to assume that a Release build has no UB - it has been caught and addressed in Debug. Also, such checks slow down execution and potentially bust the cache / instruction pipeline; although speculative execution and branch prediction will pretty much take care of it (but then also Spectre/Meltdown...).

Reply:
I don't really know what speculative execution or spectre/meltdown is but the rest makes sense. I'll get rid of `WEX_ENSURE` and replace it with a call to assert :+1: